x/dsmr/node_test.go

```diff
@@ -994,11 +1057,220 @@ func TestNode_NewBlock_IncludesChunkCerts(t *testing.T) {
 	}
 }
 
+// TestDuplicateChunksElimination tests that duplicate chunks that have appeared before are getting correctly eliminated.
+func TestDuplicateChunksElimination(t *testing.T) {
+	r := require.New(t)
+
+	node := newTestNode(t)
+
+	blk := Block{
+		ParentID:  ids.GenerateTestID(),
+		Height:    1,
+		Timestamp: 1,
+		blkID:     ids.GenerateTestID(),
+	}
+	r.NoError(node.Accept(context.Background(), blk))
+
+	_, chunkCert, err := node.BuildChunk(
+		context.Background(),
+		[]tx{
+			{
+				ID:     ids.GenerateTestID(),
+				Expiry: 4,
+			},
+		},
+		4,
+		codec.Address{},
+	)
+	r.NoError(err)
+
+	blk = Block{
+		ParentID:  ids.GenerateTestID(),
+		Height:    2,
+		Timestamp: 2,
+		blkID:     ids.GenerateTestID(),
+		ChunkCerts: []*ChunkCertificate{
+			&chunkCert,
+		},
+	}
+	r.NoError(node.Accept(context.Background(), blk))
+
+	_, err = node.BuildBlock(context.Background(), blk, 3)
```

Comment:
nit: Use `node.LastAccepted` instead of referencing `blk` for the parent to build off of